### PR TITLE
i18n error tests

### PR DESCRIPTION
**Pull request by LoinASD**
Signed-off-by: LoinASD <loinasd.dev@gmail.com>

### CODE REVIEW

File: v0.4/platform/platform/src/__tests__/i18n.test.ts [hunk 1]
```diff
@@ -38,21 +40,47 @@ describe('i18n', () => {
...
+  it('should emmit status when no loader', async () => {
+    const component = 'component'
+    const message = `${component}.id`
+
+    const checkStatus = new Status(Severity.ERROR, Code.NoLoaderForStrings, { component })
+    const eventListener = async (event: string, data: any): Promise<void> => {
+      await expect(data).toEqual(checkStatus)
+    }
+    addEventListener(PlatformEvent, eventListener)
+    await translate(message as IntlString, {})
```
Comment (z2sx) [inline v0.4/platform/platform/src/__tests__/i18n.test.ts:52]: You need separate test to check if `translate` returns expected value or check it in this test as well. Also `emit` word has one 'm'.

File: v0.4/platform/platform/src/__tests__/i18n.test.ts [hunk 2]
```diff
@@ -38,21 +40,47 @@ describe('i18n', () => {
...
-    return expect(translated).rejects.toThrowError("ERROR: platform.UnknownError")
+    const component = 'error-loader'
+    const message = `${component}.id`
+
+    const checkStatus = new Status(Severity.ERROR, Code.UnknownError, { message: errorMessage })
+    const eventListener = async (event: string, data: any): Promise<void> => {
+      await expect(data).toEqual(checkStatus)
+    }
+    addEventListener(PlatformEvent, eventListener)
+    await translate(message as IntlString, {})
```
Comment (z2sx) [inline v0.4/platform/platform/src/__tests__/i18n.test.ts:68]: Same... need to verify `translate` return here or in separate test

File: v0.4/platform/platform/src/__tests__/i18n.test.ts [hunk 3]
```diff
@@ -38,21 +40,47 @@ describe('i18n', () => {
...
+    const eventListener = async (event: string, data: any): Promise<void> => {
+      if (status === undefined) {
+        status = data
+        return
+      }
+      await expect(data).toBe(status)
+    }
+    addEventListener(PlatformEvent, eventListener)
+    await translate(message as IntlString, {})
+    await translate(message as IntlString, {})
```
Comment (z2sx) [inline v0.4/platform/platform/src/__tests__/i18n.test.ts:84]: What is this???

File: v0.4/platform/platform/src/__tests__/i18n.test.ts [hunk 4]
```diff
@@ -38,21 +40,47 @@ describe('i18n', () => {
...
+      await expect(data).toEqual(checkStatus)
+    }
+    addEventListener(PlatformEvent, eventListener)
+    await translate(message as IntlString, {})
   })
 
   it('should cache error', async () => {
-    const translated = translate('error-loader.id' as IntlString, {})
-    return await expect(translated).rejects.toThrowError("ERROR: platform.UnknownError")
+    const component = 'error-loader'
```
Comment (z2sx) [inline v0.4/platform/platform/src/__tests__/i18n.test.ts:72]: Why?

**Review by z2sx — CHANGES_REQUESTED** [review]

**Comment by z2sx** [comment]
Also where is fix or test on the 'empty object returned' problem you mentioned in the meeting?

File: v0.4/platform/platform/src/__tests__/i18n.test.ts (hunk 3 above)
Comment (LoinASD) [reply v0.4/platform/platform/src/__tests__/i18n.test.ts:84]: This is two equal requests to check if error status is cached

File: v0.4/platform/platform/src/__tests__/i18n.test.ts (hunk 2 above)
Comment (LoinASD) [reply v0.4/platform/platform/src/__tests__/i18n.test.ts:68]: done as separate test

File: v0.4/platform/platform/src/__tests__/i18n.test.ts (hunk 1 above)
Comment (LoinASD) [reply v0.4/platform/platform/src/__tests__/i18n.test.ts]: done as separate etst

File: v0.4/platform/platform/src/__tests__/i18n.test.ts (hunk 3 above)
Comment (LoinASD) [reply v0.4/platform/platform/src/__tests__/i18n.test.ts:99]: since `toBe()` checks referential identity, we can check if the status is cached

File: v0.4/platform/platform/src/__tests__/i18n.test.ts (hunk 4 above)
Comment (LoinASD) [reply v0.4/platform/platform/src/__tests__/i18n.test.ts:87]: why what?

**Review by z2sx — APPROVED** [review]
